lib/customisation.nix

```diff
@@ -159,6 +159,27 @@ rec {
       outPath = assert condition; drv.outPath;
     };
 
+    /* Add a field to a derivation’s passthru. Will refuse to add a field that is already there.
+
+       Type: { name: string, value: any } -> drv -> drv
+
+       Example:
+         (addToPassthru
+           { name = "foo"; value = "bar"; }
+           pkgs.hello).passthru.foo
+         => "bar"
+    */
+    addToPassthru =
+      # field to add
+      {name, value}:
```

Comment:
If we end up having this function I think it's much better to pass the attribute set directly as passthru's, instead of using `name` and `value`. This is more natural, allows automatic support for multiple passthru's, is more concise, and more efficient too.

Reply:
You mean to add multiple fields at once?

Reply:
No, just `addToPassthru = passthrus: ... { passthru = old.passthru or {} // passthrus; }`

Reply:
So add multiple fields at once?

Reply:
I guess yeah :)

Reply:
Well it doesn't have to be used for multiple fields of course